kubernetes/kubernetes/src/main/java/org/arquillian/cube/kubernetes/impl/locator/DefaultKubernetesResourceLocator.java

```diff
@@ -25,6 +33,41 @@ public URL locate() {
         return null;
     }
 
+    @Override
+    public URL locateFromTargetDir() {
+        for (String resource : getResourceNames()) {
+            for (String suffix : getAllowedSuffixes()) {
```

Comment:
This looks to me a bit too heavy for the problem we are aiming to solve. Could you try to just use `Files.walk` with filtering based on what ` getResourceNames()` and ` getAllowedSuffixes()` contain? These can be sets which we can use for the conditions.